src/Native/Windows/clrcompression/CMakeLists.txt

```diff
@@ -77,7 +77,7 @@ else()
     )
 endif()
 
-if ($ENV{__VSVersion} STREQUAL vs2015)
+if ($ENV{__VSVersion} STREQUAL vs2017)
```

Comment:
Nit: this could be unconditionally feature detected like: `check_symbol_exists(vsnprintf stdio.h HAVE_VSNPRINTF)`

Reply:
I don't know much about cmake.

```
if (check_symbol_exists(vsnprintf stdio.h HAVE_VSNPRINTF))
    add_definitions(-DHAVE_VSNPRINTF)
endif ()
```

Is that correct?

Reply:
Correct, though I am not sure about the boolean nature of cmake functions. :)

```cmake
check_symbol_exists(vsnprintf stdio.h HAVE_VSNPRINTF)
if (HAVE_VSNPRINTF) # HAVE_VSNPRINTF is CMAKE variable, compiler doesn't know about it yet
    add_definitions(-DHAVE_VSNPRINTF) # this is for the compiler
endif ()
```

For the projects that have config.h.in and `configure_file()` in their CMAKE, `check_symbol_exists` would be enough in CMAKE and a corresponding `cmakedefine` will go in the config file (like `pal_crypto_config.h.in`).
In clrcompression, there is no config file, therefore add_definition is required.

Reply:
thanks a lot 👍

Reply:
The include is also missing in this file, so `include (CheckSymbolExists)` one line above (or at the top of file, doesn't matter).